_adr/92/index.adoc

```diff
@@ -0,0 +1,104 @@
+* a single place with shared logic for aggregating and serving data
+* reduce the interactions over the network for providing a given functionality
+* easy testing/mocking setup for UI/CLI
+
+## Non-goals
+
+* provide a generic API layer for consuming the current services APIs
+* provide a generic queryable layer for the current services APIs
+
+## Current situation
+
+In the current architecture both UI and CLI are reaching out both Control Plane and Data Plane REST API endpoints.
+
+image::current_architecture.png[Current Architecture]
+
+
+## Proposal
+
+[NOTE]
+*Assumption* as a first iteration we are assuming that the CLI needs can be covered by a strict subset of the functionalities needed for the UI.
+
+We propose to integrate *one* server side component commonly known as https://samnewman.io/patterns/architectural/bff/[BFF] (Backend For Frontend).
+
+image::with_BFF.png[Proposed Architecture]
+
+The service is implemented using Java and Quarkus to achieve maximum collaboration across the current teams and align with Red Hat tech strategy.
+The API layer is designed using API Designer(RHOAD) and conforming to the OpenAPI 3+ specification to achieve maximum collaboration across different teams and members with various technological background.
+The development is driven by a "Contract First" approach leveraging the [Apicurio Codegen Quarkus Extension](https://github.com/Apicurio/apicurio-codegen/tree/main/quarkus-extension) and re-using the Java SDKs(based on Kiota) provided by the DevExp Team.
+The BFF have to to be stateless so that it can easily be scaled up and down according to the needs.
+All of the exposed endpoints are going to perform 1+ REST APIs calls to the underlying services and provide back the result.
+
+### Threat model
+
+Being it a publicly accessible service the generic service Threat model applies, we should ensure stability of the platform by having all the good practices in place:
+
+* Rate Limiting
+* Circuit Breaking
+* Caching
+
+Regarding multitenancy, it can be achieved, in the first prototype by re-using the Access Token for the initial API request to issue the following requests to the backend services.
+
+## Alternatives Considered / Rejected
+
+### GraphQL
+
+Is a well known technical platform for implementing BFFs.
+
+* Lack of expertise
+* We don't want a generic API
+* Challenges in caching
+* Poor support for "commands"(as opposed to queries)
+
+### Expand the current services APIs
+
+We want the current microservices to stay vertical and opinionated toward a minimal and curated API that directly maps the functionality of the service without having to deal with the integration challenges.
+
+## Challenges
+
+* Caching while guaranteeing multitenancy
+* Shared ownership of the component across different teams
+* Avoid API proliferation after the first BFF is in place
+* Potential lack of control over APIs being aggregated
```

Comment:
Another challenge worth noting is that with the notion of private data planes the control plane may not have access to the back-end APIs or data sources.

Reply:
added